### PR TITLE
Persist configuration in localstorage

### LINKED ISSUE #127: Persist some configuration data
It's annoying to set is nickname each time we connect, same for notification sound, etc.

It would be nice if this data could be stored in the localstorage for instance.

### PR DESCRIPTION
**Pull request by jrmi**
Closes #127

### CODE REVIEW

**Comment by seripap** [comment]
Just a thought on this- I'd disable this feature by default and leave it up to the user to persist or not. This will keep with the core mission of Darkwire being an ephemeral chat messaging system

File: client/src/utils/persistence.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/**
+ * Handle localStorage persistence
+ */
+
+import { debounce } from 'lodash';
+import { getTranslations } from 'i18n';
+
+export const loadPersistedState = () => {
+  const state = {};
+  const language = localStorage.getItem('language');
```
Comment (seripap) [inline client/src/utils/persistence.js:10]: Perhaps merge to a single object that is stringified, that way you'd only have to getItem on something like a `settings` node.

**Comment by jrmi** [comment]
> Just a thought on this- I'd disable this feature by default and leave it up to the user to persist or not. This will keep with the core mission of Darkwire being an ephemeral chat messaging system

Agreed. I was thinking about something like that in settings.

File: client/src/utils/persistence.js (hunk 1 above)
Comment (jrmi) [reply client/src/utils/persistence.js]: Fixed!

**Comment by jrmi** [comment]
> Just a thought on this- I'd disable this feature by default and leave it up to the user to persist or not. This will keep with the core mission of Darkwire being an ephemeral chat messaging system

Also fixed. The persistence is now optional.

File: client/src/utils/persistence.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  // We need prev state to compare
+  if (prevState) {
+    const {
+      user: { username },
+      app: { notificationIsEnabled, soundIsEnabled, persistenceIsEnabled, language },
+      room: { id: roomId },
+    } = state;
+
+    if (prevState.room.id !== roomId && roomId) {
+      if (!prevState) console.log('room id set');
```
Comment (seripap) [inline client/src/utils/persistence.js:58]: remove console

**Review by seripap — CHANGES_REQUESTED** [review]
Small changes

**Pull request merged**